### PR TITLE
refactor: rename `namespace` to `mod`

### PR DESCRIPTION
**Pull request by jaschdoc**
This PR removes the `namespace` keyword and `/` namespace separator.

Closes https://github.com/flix/flix/issues/4800

### CODE REVIEW

**Comment by magnus-madsen** [comment]
We should also be using `.` as separator instead of `/`. I suggest to do this in the same PR.

**Comment by jaschdoc** [comment]
> We should also be using `.` as separator instead of `/`. I suggest to do this in the same PR.

Working on it! :smile:

**Comment by jaschdoc** [comment]
Should these `/` separators also be `.` (in the key part of the association, of course, i.e., `Fixpoint._ -> LocalResource.get(...)`)?
https://github.com/flix/flix/blob/a9b17efdb3a9cb9b810d82361023f8ba89abc942/main/src/ca/uwaterloo/flix/api/Flix.scala#L213

File: main/src/ca/uwaterloo/flix/language/phase/Parser.scala [hunk 1]
```diff
@@ -162,7 +162,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
   object Declarations {
 
     def Namespace: Rule1[ParsedAst.Declaration.Namespace] = rule {
-      optWS ~ SP ~ (keyword("namespace") | keyword("mod")) ~ WS ~ Names.Namespace ~ optWS ~ '{' ~ UsesOrImports ~ Decls ~ optWS ~ '}' ~ SP ~> ParsedAst.Declaration.Namespace
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Parser.scala:165]: Could we restore the Parser; just to give people a bit more time to upgrade themselves. Thanks!

**Comment by jaschdoc** [comment]
@magnus-madsen reverted parser changes

**Comment by magnus-madsen** [comment]
> Reverted parser changes

Thanks,- can you put them in a separate PR? :)

**Comment by magnus-madsen** [comment]
Excellent. Thanks!

**Pull request merged**